### ISSUE TITLE
though 'ↂ' is a valid identifier in ECMAScript 5.1, unexpected character error thrown when building

### ISSUE THREAD

**Issue by 0a-** [meteor/meteor#3360]
My app runs fine locally. But when building it, an unexpected character error would be thrown because ↂ is used in a variable name. Since ↂ (unicode number: U+2182) is a valid identifier in ECMAScript 5.1, it would be really cool if Meteor can accept variables with names like ↂωↂ (totally legit identifier) when building the app for deployment. Thanks!  

A variable with the identifier ಠ_ಠ goes through the building without an error. :+1:

**Comment by glasser** [comment]
Looks like this is an issue with UglifyJS.  Can you raise the issue on [their repo](https://github.com/mishoo/UglifyJS2) (or is it fixed on the latest version there)?  Then we can upgrade to the fixed version

```
[[[[[ /private/tmp/asd ]]]]]

=> Started proxy.
                                                     
/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/fibers/future.js:173
						throw(ex);
						      ^
Error: Unexpected character 'ↂ' (line: 1, col: 16, pos: 16)
    at Object.Future.wait (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/fibers/future.js:323:16)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:556:31
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:554:9
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at enterJob (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:317:26)
    at Object.forkJoin (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:531:3)
    at [object Object]._.extend.minifyJs (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:812:28)
    at [object Object]._.extend.make (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:505:12)
    at makeClientTarget (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:1988:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/bundler.js:2032:20
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /Users/glasser/Projects/Meteor/meteor/tools/bundler.js:2031:7
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:264:13
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:257:29
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:255:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:246:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at Object.capture (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:245:19)
    at Object.exports.bundle (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:1975:31)
    at bundleApp (/Users/glasser/Projects/Meteor/meteor/tools/run-app.js:508:34)
    at [object Object]._.extend._runOnce (/Users/glasser/Projects/Meteor/meteor/tools/run-app.js:544:35)
    at [object Object]._.extend._fiber (/Users/glasser/Projects/Meteor/meteor/tools/run-app.js:771:28)
    at /Users/glasser/Projects/Meteor/meteor/tools/run-app.js:366:12
    - - - - -
    at Object.Future.wait (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/fibers/future.js:323:16)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:556:31
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:554:9
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at enterJob (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:317:26)
    at Object.forkJoin (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:531:3)
    at _minify (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:925:18)
    at /Users/glasser/Projects/Meteor/meteor/tools/bundler.js:814:16
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:537:23
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at enterJob (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:317:26)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:536:24
    at runWithEnvironment (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:139:21)
    at /Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:146:14
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:549:11
    - - - - -

Error
    at new JS_Parse_Error (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:196:18)
    at js_error (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:204:11)
    at parse_error (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:306:9)
    at Object.next_token [as input] (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:551:9)
    at next (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:649:25)
    at /Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:765:42
    at /users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:710:24
    at block_ (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:984:20)
    at ctor.body (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:957:25)
    at function_ (/Users/glasser/Projects/Meteor/meteor/.meteor/isopackets/minifiers/npm/minifiers/node_modules/uglify-js/lib/parse.js:962:15)
    at Object.Future.get (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/fibers/future.js:124:86)
    at Object.Future.wait (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/fibers/future.js:323:16)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:556:31
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/glasser/Projects/Meteor/meteor/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:554:9
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at enterJob (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:317:26)
    at Object.forkJoin (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:531:3)
    at _minify (/Users/glasser/Projects/Meteor/meteor/tools/bundler.js:925:18)
    at /Users/glasser/Projects/Meteor/meteor/tools/bundler.js:814:16
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:537:23
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:352:18
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:345:34
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:343:23
    at [object Object]._.extend.withValue (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:115:14)
    at enterJob (/Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:317:26)
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:536:24
    at runWithEnvironment (/Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:139:21)
    at /Users/glasser/Projects/Meteor/meteor/tools/fiber-helpers.js:146:14
    at /Users/glasser/Projects/Meteor/meteor/tools/buildmessage.js:549:11
```

**Comment by 0a-** [comment]
Alright, hope they would add in more utf-8 support in the next version. (Just tested ↂ in the [online demo](http://lisperator.net/uglifyjs/#demo), and got an unexpected char error immediately)

**Comment by glasser** [comment]
Looks like a PR to fix this got merged in uglify but hasn't been released yet.  People who care about this bug, let us know when there's a released version :)

**Comment by glasser** [comment]
Also we should see if that fix fixes #2877 as well.
